### DISCUSSION TITLE
Images not being presented on some trials

### DISCUSSION THREAD

**Discussion by Quinlanptq** [jspsych/jsPsych#3141] [Q&A]
On each trial in my experiment I present a fixation 

var fixation = {
    type: jsPsychHtmlKeyboardResponse,
    choices: "NO_KEYS",
    stimulus: '+',
    trial_duration: 250,
    post_trial_gap: 300,
};

and then immediately an image

var trial = {
    type: jsPsychImageKeyboardResponse,
    choices: ['d','k'],
    
    stimulus: function () {
    
        return (
               jsPsych.timelineVariable("the_image")
            );
        },
    stimulus_duration: 1000,
    trial_duration: 2000,
    post_trial_gap :1000,
    data: {
        
              Got_cha:'Yes',
        
            }
    };    

The problem that I have is that it is possible to press a key before the image is actually presented. 

and responses rts are recorded as null

Can anyone tell why this happening and how do I stop it?
Thanks

Philip.

**Comment by jodeleeuw** [comment]
Hi @Quinlanptq,

Are you preloading the images?

Relevant docs: https://www.jspsych.org/7.3/overview/media-preloading/

**Comment by Quinlanptq** [comment]
I think so here is the critical code - this of course may not be doing what I think it should be doing. I posted an early version of the code on show and tell
Thanks

Philip.

```js
for (var bc = 0; bc<4; bc++){
    
    this_block = the_blocks[bc];
    factors = {
        the_image : this_block
    };
    
    factorial_values = jsPsych.randomization.factorial(factors);
    

    trials_with_variables = {
        timeline: [fixation, trial],
        timeline_variables: factorial_values
    };
    
    preload_b = {
        type : jsPsychPreload,
        auto_preload: true,
        trials : trials_with_variables
    };
    
    big_line.push(preload_b);
    big_line.push(trials_with_variables);
    big_line.push(end_of_block);
  
};

    big_line.push(cursor_on);
    big_line.push(debrief_block);
    big_line.push(end_of_exp);
    jsPsych.run(big_line);
```

**Comment by jodeleeuw** [comment]
The preload plugin is unable to automatically detect which timeline variables are images, so you can't use the `trials` parameter to load those values in this case. Is there a variable that has a list of all the image files in the block, stored as an array? If so, you can pass that array to the `images` parameter of the preload plugin to accomplish the same goal.

**Comment by Quinlanptq** [comment]
Thanks Josh 

re-wrote ny pre_load_b var as follows

    preload_b = {
        type : jsPsychPreload,
        auto_preload: true,
        images: this_block,
        trials : trials_with_variables
    };

and this works fine - I thought the images were being handled by the trials variable
Best
Philip.
